### PR TITLE
perf(protorev): send rewards to dev acc once per epoch

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #7518

## What is the purpose of the change

The protorev module currently calculates profits and sends the dev reward cut after every backrun. This adds unnecessary read/write overhead, causing measurable detriment to sync speed.

This PR tracks the difference between the last epoch's protorev module account to the balance at the current epoch, calculates the dev rewards once, and does a single send.

### CODE REVIEW

**Comment by czarcas7ic** [comment]
https://github.com/osmosis-labs/osmosis/pull/7509

^ this PR should be merged before this one, so it can be properly wired

File: x/protorev/keeper/hooks.go [hunk 1]
```diff
@@ -221,3 +242,48 @@ func (k Keeper) CompareAndStorePool(ctx sdk.Context, poolId uint64, baseDenom, o
...
+	err = k.DistributeProfit(ctx, profit)
+	if err != nil {
+		return fmt.Errorf("failed to send developer fee: %w", err)
+	}
+
+	return nil
+}
+
+// CalculateCurrentProfit retrieves the current balance of the protorev module account and filters for base denoms.
+func (k Keeper) CurrentBaseDenomProfits(ctx sdk.Context) (sdk.Coins, error) {
```
Comment (czarcas7ic) [inline x/protorev/keeper/hooks.go:267]: TODO: Write tests for these two

**Review by davidterpay — APPROVED** [review]

**Comment by czarcas7ic** [comment]
Will get test in and then merge, thanks all

**Comment by czarcas7ic** [comment]
Im actually seeing the multi-denom test here already https://github.com/osmosis-labs/osmosis/blob/2978b5e80019ed83bc836ef5ce3e36e375aa557d/x/protorev/keeper/hooks_test.go#L864-L873

@ValarDragon lmk if that suffices.

**Pull request merged**